### DISCUSSION TITLE
Get authentication from AWS metadata

### DISCUSSION THREAD

**Discussion by alejobs** [saleweaver/python-amazon-sp-api#141] [Ideas]
Hi,

I we are running this project in an AWS resource, such as EC2, ECS docker, Lambda, etc. we may leverage AWS metadata to obtain AWS credentials so not to provide them manually.

We could use this library: https://github.com/DavidMuller/aws-requests-auth (see section named: Using Boto To Automatically Gather AWS Credentials).

Thanks.

**Comment by saleweaver** [comment]
Hi,

Simply set the required values as explained in the doc's environment variable section

**Reply by alejobs** [reply]
Sorry, but this is not a solution. I need not to provide credentials but grant EC2 metadata to get assigned role and use this IAM role to sign the request. It does not make sense to have to generate a secret & access key in the same AWS account you are executing this code.
